helper/restore_helper.go

```diff
@@ -239,6 +243,11 @@ func doRestoreAgent() error {
 			}
 		}
 
+		if tableOid == skipOid {
+			logVerbose(fmt.Sprintf("Oid %d, Batch %d: skip due to skip file\n", tableOid, batchNum))
+			goto LoopEnd
```

Comment:
Once we go to the LoopEnd, we would encounter [this block](https://github.com/arenadata/gpbackup/blob/21ba1a1c96c8ed2ee5cd27b201b0b2c2547eed2e/helper/restore_helper.go#L359):
```
		// On resize restore reader might be nil.
		if !*singleDataFile && !(*isResizeRestore && contentToRestore >= *origSize) {
			if errPlugin := readers[contentToRestore].waitForPlugin(); errPlugin != nil {
				if err != nil {
					err = errors.Wrap(err, errPlugin.Error())
				} else {
					err = errPlugin
				}
			}
		}
```
Did you test resize-restore with plugin on your code (and with contentToRestore < *origSize)?
It looks that, once we start skipping batchNum for a particular table OID, we might call `waitForPlugin()` without prior getting new restore  [data reader](https://github.com/arenadata/gpbackup/blob/21ba1a1c96c8ed2ee5cd27b201b0b2c2547eed2e/helper/restore_helper.go#L259) .
Please double check that we are on a safe side here.

Reply:
Checked one more time and added test. Still looks okay to me.